results-processor/processor.py

```diff
@@ -242,6 +243,26 @@ def create_run(self, run_id, labels, uploader, callback_url=None):
             callback_url)
         assert self.test_run_id
 
+    def update_status(self, run_id, stage, error=None, callback_url=None):
+        assert stage, "stage cannot be empty"
+        if int(run_id) == 0:
+            _log.error('Cannot update run status: missing run_id')
+            return
+        if callback_url is None:
+            callback_url = config.project_baseurl()
+        parsed_url = urlparse(callback_url)
+        api = '%s://%s/api/status/update' % (parsed_url.scheme,
```

Comment:
Normally REST architecture here for and update would be a PUT to `/api/status/{id}`. See `PATCH` comment above.